R/ArchiveTuning.R

```diff
@@ -94,17 +146,35 @@ ArchiveTuning = R6Class("ArchiveTuning",
     #' @param ... (ignored).
     print = function() {
       catf(format(self))
-      print(self$data[, setdiff(names(self$data), c("x_domain", "uhash")), with = FALSE], digits=2)
+      print(as.data.table(self, unnest = NULL, exclude_columns = c("x_domain", "uhash")), digits=2)
     }
-  ),
+  )
+)
 
-  active = list(
+#' @export
+as.data.table.ArchiveTuning = function(x, ..., unnest = "x_domain", exclude_columns = "uhash", measures = NULL) {
+  if (nrow(x$data) == 0) return(data.table())
+  # default values for unnest and exclude_columns might be not present in archive
+  if ("x_domain" %nin% names(x$data)) unnest = unnest[unnest %nin% "x_domain"]
```

Comment:
```suggestion
  if ("x_domain" %nin% names(x$data)) unnest = setdiff(unnest, "x_domain")
```
Easier to read? Maybe a nonsecond faster?